src/storage/page_buffer.c

```diff
@@ -11102,6 +11110,30 @@ pgbuf_ordered_fix_release (THREAD_ENTRY * thread_p, const VPID * req_vpid, PAGE_
       req_watcher->curr_rank = PGBUF_ORDERED_HEAP_HDR;
     }
 
+  for (i = 0; i < saved_pages_cnt; i++)
+    {
+      if (ordered_holders_info[i].prevent_dealloc)
+	{
+	  /* we need to remove prevent deallocate. */
+	  PGBUF_BUFFER_HASH *hash_anchor = &pgbuf_Pool.buf_hash_table[PGBUF_HASH_VALUE (&ordered_holders_info[i].vpid)];
+	  bufptr = pgbuf_search_hash_chain (hash_anchor, &ordered_holders_info[i].vpid);
+
+	  if (bufptr == NULL)
+	    {
+	      /* oops... no longer in buffer?? */
+	      assert (false);
+	      continue;
+	    }
+	  if (bufptr->avoid_dealloc_cnt == 0)
+	    {
+	      /* oops... deallocate not prevented */
+	      assert (false);
+	      continue;
+	    }
+	  ATOMIC_INC_32 (&bufptr->avoid_dealloc_cnt, -1);
+	}
+    }
+
```

Comment:
I think we have manage mutexes for hash_anchor and bcb, and would like propose the change as follows:

```C
  for (i = 0; i < saved_pages_cnt; i++)
    {
      if (ordered_holders_info[i].prevent_dealloc)
        {
          /* we need to remove prevent deallocate. */
          PGBUF_BUFFER_HASH *hash_anchor = &pgbuf_Pool.buf_hash_table[PGBUF_HASH_VALUE (&ordered_holders_info[i].vpid)];
          bufptr = pgbuf_search_hash_chain (hash_anchor, &ordered_holders_info[i].vpid);

          if (bufptr == NULL)
            {
              /* oops... no longer in buffer?? */
              assert (false);
              pthread_mutex_unlock (&hash_anchor->hash_mutex);
              continue;
            }

          if (bufptr->avoid_dealloc_cnt <= 0)
            {
              /* oops... deallocate not prevented */
              assert (false);
            }
          else
            {
              ATOMIC_INC_32 (&bufptr->avoid_dealloc_cnt, -1);
            }

          PGBUF_BCB_UNLOCK (bufptr);
        }
    }
```

We may have to also confirm the case: `bufptr == NULL`
This case means that bcb was victimized by the other. It is very unlikely, but I'm not 100% sure. 
Can you confirm it?